### ISSUE TITLE
NSIS (untgz.dll) issue with len(filename) == 100

### ISSUE THREAD

**Issue by ThomasLecocq** [ContinuumIO/anaconda-issues#3264]
Hi,

With @megies we are creating anaconda+obspy installers for all platforms (see https://github.com/obspy/obspy/pull/1818, https://github.com/conda/constructor/issues/92 and https://github.com/conda/constructor/issues/93). 

At installation time (running the setup.exe on a Windows machine), we are encountering a problem with untgz.dll when the file (actually, path+filename) to extract is *exactly* 100 char long. Smaller file names get extracted normally, and longer filenames are handled by the tgz_extract properly, for example (extracted from the log of the installation in the GUI):

```
tgz_extract: using GNU long filename [Lib/site-packages/obspy/taup/tests/data/TauP_test_data/taup_pierce_-mod_isp91_ph_P_-h_10_-evt_-45_-50_-sta_-80_-50]
```

while this:

```
Writing Lib/site-packages/obspy/io/mseed/tests/data/one_record_time_corr_applied_but_time_corr_is_zero.mseed0000666
```

is bad, the filename ending after .mseed, and the `0000666` must be some random crap coming from hell or else?

```python
In [5]: a = "Lib/site-packages/obspy/io/mseed/tests/data/one_record_time_corr_applied_but_time_corr_is_zero.mseed"

In [6]: len(a)
Out[6]: 100
```

The conda `nsis 3.0.1` package includes the `untgz` plugin, but not the one mentionned on NSIS website (http://nsis.sourceforge.net/UnTGZ_plug-in) as this one is way too old (1.0.18 from 2010), but rather, I guessed after much searching on the web, the `master` version of this https://github.com/mingwandroid/nsis-untgz (untagged, but mentionned 1.0.21.0 in https://github.com/mingwandroid/nsis-untgz/commit/5c814c5f2c8a9a14e3a6ddd3e594fcc76db5b86a)

Our guess with @megies is that you are using this NSIS code to create the Anaconda and Miniconda installers too, so this Issue could be quite more critical than just our test file.

Cheers,
Tom

**Comment by megies** [comment]
> when the file (actually, path+filename) to extract is exactly 100 char long

At least that's the one case we encountered a problem with.. we can't exclude with certeinty that other lengths are not affected as well..

> Our guess with @megies is that you are using this NSIS code to create the Anaconda and Miniconda installers too, so this Issue could be quite more critical than just our test file.

I totally agree.. from what we figured, this problem might hit any Anaconda package that happens to have a file path with exactly 100 characters, leading to very obscure and hard to debug problems.

**Comment by megies** [comment]
CC @mingwandroid @tpn @ilanschnell since you seem to be involved in `nsis-ungtz`..

**Comment by megies** [comment]
This might be related as well: https://github.com/ilanschnell/untgz/commit/9f72e0478a617312b841dcf57b470c2694ae6da0

**Comment by mingwandroid** [comment]
I'll take a look, thanks.

**Comment by mingwandroid** [comment]
Fixed, thanks @jjhelmus and sorry I forgot about this one!

**Comment by megies** [comment]
Thanks for the fix. Any chance to get a link to the fix (commit or PR)?

**Comment by jjhelmus** [comment]
Fix in mingwandroid/nsis-untgz#1

**Comment by megies** [comment]
Thanks @jjhelmus!
